kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/config/UnresolvedKeyPolicy.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.config;
+
+/**
+ * This policy determines what the Filter should do with records when we cannot
+ * resolve a Key for that record.
+ */
+public enum UnresolvedKeyPolicy {
+    /**
+     * Passthrough the record unencrypted. This is the default policy, it allows unencrypted and encrypted topics to be handled by
+     * the same Filter.
+     */
+    PASSTHROUGH_UNENCRYPTED,
+    /**
+     * Reject the record. This will cause the entire ProductRequest to be rejected and an appropriate error message will be returned
+     * to the client. This is a safer policy if you know that all traffic sent to the virtual cluster should be encrypted.
+     */
+    REJECT
```

Comment:
I think we should default to fail closed 
1. Its better security posture 
2. Its recoverable. If we reject records they can be re-sent, we we pass through and we should have encrypted we have plain text in a supposedly encrypted topic and no mechanism to fix that.

Reply:
ARRRGHGH right, I see why you default to `PASSTHROUGH_UNENCRYPTED`  as we still don't know what topics are supposed to be encrypted vs the missing key. 

Ok until we fix that we need to default to `PASSTHROUGH_UNENCRYPTED`.